xonsh/prompt/gitstatus.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""informative git status for prompt"""
+
+import os
+import builtins
+import subprocess
+
+
+def check_output(*args, **kwargs):
+    kwargs.update(dict(env=builtins.__xonsh_env__.detype(),
+                       stderr=subprocess.DEVNULL,
+                       timeout=builtins.__xonsh_env__['VC_BRANCH_TIMEOUT'],
+                       universal_newlines=True
+                       ))
+    return subprocess.check_output(*args, **kwargs)
+
+_DEFS = {
```

Comment:
this can be a function decorated with `lazyobject` from [lazyasd](https://github.com/xonsh/xonsh/blob/master/xonsh/lazyasd.py#L126) that returns the dict like:
```
@lazyobject
def _DEFS():
    return {...}
```